native/more/relationship-list.react.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+
+import type { NavigationRoute } from '../navigation/route-names';
+import type { AppState } from '../redux/redux-setup';
+import type { MoreNavigationProp } from './more.react';
+
+import React from 'react';
+import { View, Text, FlatList } from 'react-native';
+
+import { connect } from 'lib/utils/redux-utils';
+import { type UserInfo } from 'lib/types/user-types';
+
+import { AddFriendsModalRouteName } from '../navigation/route-names';
+import { styleSelector } from '../themes/colors';
+
+import RelationshipListItem from './relationship-list-item.react';
+import RelationshipListAddButton from './relationship-list-add-button.react';
+
+type Props = {|
+  navigation: MoreNavigationProp<'FriendList'>,
```

Comment:
`RelationshipList` is used for two different components, but here you've defined it specifically for `FriendList`. In this case you aren't using the navigation params, so it should be fine to just leave the route unspecified instead:

```js
  navigation: MoreNavigationProp<>,
  route: NavigationRoute<>,
```

Actually, I've realized that `NavigationRoute` doesn't have a default type param. For this to work, you'll have to update it to have a default type parameter `$Keys<ScreenParamList>`, similar to how `MoreNavigationProp` works.

Reply:
I left it unspecified, but I think it's not a good idea for `route` prop. It would work for `navigation` since `routeName` must be always provided when `navigation.navigate()` is called so Flow can infer the `params` type for that route and keep it strict. However, `route` prop is used for getting the navigation params for the current screen so when we would have `NavigationRoute<>` with the unspecified route name it wouldn't be possible to infer the `params` type. `props.route.params` will be `any` in that case. I don't like having unsafe types in the code.

I'm gonna use RelationshipList as a component that would be used in two separate screen components: FriendListScreen and BlockListScreen. What do you think?

Reply:
1. You're confusing a couple different `RouteName`s. The `DestinationRouteName` in `navigate` is parameterized at invocation, and can be any route name in `GlobalParamList`. The `RouteName` that `NavigationProp` is parameterized on is the `RouteName` of the current route, and it is used only to type `setParams`. In our case we aren't using `setParams`, so it doesn't matter.
2. Are you sure `this.props.route.params` is `any`? I think it should be the union of all the params for every screen in `MoreParamList`. I think it should be sound, but Flow has a lot of quirks - let me know if you're seeing it be typed as `any`. (You can examine the type of a specific property by extracting it `const { params } = this.props.route;` and then using `flow type-at-pos` on it)
3. Using `RelationshipList` for two screens sound like a fine plan. If you end up needing params, as long as they're the same between the two screens it should still be possible to type correctly like `MoreNavigationProp<'FriendList' | 'BlockList'>` I think

Reply:
1. That it what I meant. `navigation.navigate` requires a `routeName` argument so Flow knows what is the type of the `params` of that route based on `GlobalParamList` and `routeName` passed to `navigate` function.
2. You're right. It's a union type. I had to restart flow server to see that. Previously I had no flow errors and `any` for `params`.
3. `MoreNavigationProp<'FriendList' | 'BlockList'>` works fine 👌